MMDevice/MMDevice.h

```diff
@@ -1428,4 +1445,307 @@ namespace MM {
+       *                  The last two dimensions should be "Y" and "X" if we are dealing with images
+       */
+      virtual int ConfigureDimension(const char* handle, int dimension,
+                                     const char* name, const char* meaning) = 0;
+
+      /**
+       * \brief   Configures a coordinate's properties
+       *
+       * \param   handle Dataset handle
+       * \param   dimension The dimension index
+       * \param   coordinate The coordinate index within the dimension
+       * \param   name Name for the coordinate
+       * \return  Status code indicating success or failure
+       */
+      virtual int ConfigureCoordinate(const char* handle, int dimension,
+         int coordinate, const char* name) = 0;
+
+      /**
+       * \brief   Closes an opened dataset
+       *
+       * \param   handle Dataset handle (becomes invalid after closing)
+       * \return  Status code indicating success or failure
+       */
+      virtual int Close(const char* handle) = 0;
+
+      /**
+       * \brief Checks if a dataset is currently open
+       *
+       * \param handle Dataset handle
+       * \return true if dataset is open, false otherwise
+       */
+      virtual bool IsOpen(const char* handle) = 0;
+
+      /**
+       * \brief   Checks if a dataset is read-only
+       * \details Loaded datasets are read-only and cannot accept new images.
+       *          Only new datasets accept images, until they are closed.
+       * \param   handle Dataset handle
+       * \return  true if dataset is read-only, false if it can accept new images
+       */
+      virtual bool IsReadOnly(const char* handle) = 0;
+
+      /**
+       * \brief   Loads an existing dataset
+       * \details Loaded datasets are immutable and cannot accept new images.
+       *          Implementation may use lazy loading for efficiency.
+       *
+       * \param   path Path to the dataset
+       * \param   [out] handle Output parameter for the dataset handle
+       * \return  Status code indicating success or failure
+       */
+      virtual int Load(const char* path, char* handle) = 0;
+
+      /**
+       * \brief   Checks if the device can load a dataset at the specified path
+       * \details This will be used by the caller to quickly determine which driver to use
+       *          for a given dataset path. The implementation should ideally be very fast
+       *          and not open any files.
+       * \param   path Path to the dataset
+       * \return  true if the device can load the dataset, false otherwise
+       */
+      virtual bool CanLoad(const char* path) = 0;
+
+      /**
+       * \brief Gets the progress of the current operation
+       *
+       * \param handle Dataset handle
+       * \return Progress value (0-100), or -1 if idle or not implemented
+       * \note Assumes single operation execution at a time
+       */
+      virtual int GetProgress(const char* handle) = 0;
```

Comment:
Is this intended to be called on a second thread while the first thread is making the call to perform an operation?